topdown/eq.go

```diff
@@ -32,6 +32,8 @@ func evalEqGround(t *Topdown, a ast.Value, b ast.Value, iter Iterator) error {
 	}
 	if ast.Compare(a, b) == 0 {
 		return iter(t)
+	} else if errGroundNotEqual {
+		return fmt.Errorf("scalars did not unify: %d and %d", a, b)
```

Comment:
f(1, y) should only unify with the first decleration. If it's unifying with both, there's a bug somewhere.